indexer/feature_decl.cpp

```diff
@@ -20,3 +20,13 @@ string DebugPrint(feature::EGeomType type)
   case EGeomType::GEOM_AREA: return "GEOM_AREA";
   }
 }
+
+pair<FeatureID::MwmName, FeatureID::MwmVersion> FeatureID::GetMwmNameAndVersion() const
+{
+  if (!IsValid())
+    return {"INVALID", 0};
```

Comment:
Я согласен с Юрой, надо чуток переделать.

> On Mar 23, 2016, at 22:36, ygorshenin <notifications@github.com> wrote:
> 
> In indexer/feature_decl.cpp <https://github.com/mapsme/omim/pull/2446#discussion_r57223214>:
> 
> > @@ -20,3 +20,13 @@ string DebugPrint(feature::EGeomType type)
> >    case EGeomType::GEOM_AREA: return "GEOM_AREA";
> >    }
> >  }
> > +
> > +pair<FeatureID::MwmName, FeatureID::MwmVersion> FeatureID::GetMwmNameAndVersion() const
> > +{
> > +  if (!IsValid())
> > +    return {"INVALID", 0};
> NOT LGTM
> Это некорректно возвращать какую то непустую строку. Где гарантии что это не просочится в UI или не повлияет на какие то алгоритмы. Я бы еще понял если бы где то в районе storage была заведена константа для невалидных имен Mwm (пустая строка вроде как), но вот так несфига возвращать непустую строку для невалидныз имен это не вариант. И зачем метод для пары имя версия? Почему не завести два метода, один для имени второй для версии, или просто не использовать MwmInfo?
> 
> Короче, я против такого PR.
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2446/files/73d25ca456e5c6398ea751ba125c82a456a098ec#r57223214>